src/drivers/frsky_telemetry/sPort_data.c

```diff
@@ -64,13 +67,24 @@ static int global_position_sub = -1;
 static int battery_status_sub = -1;
 static int vehicle_status_sub = -1;
 static int gps_position_sub = -1;
+static int vehicle_attitude_sub = -1;
+static int mission_result_sub = -1;
+static int mavlink_log_sub = -1;
 
 static struct sensor_combined_s *sensor_combined;
 static struct vehicle_global_position_s *global_pos;
 static struct battery_status_s *battery_status;
 static struct vehicle_status_s *vehicle_status;
 static struct vehicle_gps_position_s *gps_position;
+static struct vehicle_attitude_s *vehicle_attitude;
+static struct mission_result_s *mission_result;
+static struct mavlink_log_s *mavlink_log;
 
+//FIFO for mavlink messages
+#define FIFO_ELEMENTS 636 //12*52, so up to 12 messages, worst case.
+#define FIFO_SIZE (FIFO_ELEMENTS + 1)
+uint8_t fifo[FIFO_SIZE];
```

Comment:
This will eat the RAM globally, even if its not used. And its a pretty big allocation. This should be done inside the driver, not globally, and 600 bytes seems way too much. Try to cut it down to 3-4 messages.